src/devices/src/legacy/serial.rs

```diff
@@ -199,8 +199,16 @@ impl Serial {
                     }
                 } else {
                     if let Some(out) = self.out.as_mut() {
-                        out.write_all(&[value])?;
-                        METRICS.uart.write_count.inc();
+                        let res = out.write(&[value]);
```

Comment:
We only write one byte at a time. `&[value]` is a slice of one byte. So partial writes aren't possible here.

But following Alin's observation, can we get `Ok(0)` in case of HUP or some other reason?

Reply:
The [write man page](https://www.man7.org/linux/man-pages/man2/write.2.html) is inconclusive: 

"If count is zero and fd refers to a file other than a regular file, the results are not specified."

After reading the man page, I don't see any scenario in which we would get an `Ok(_)` but it was actually an error. I made this statement with the following hypothesis:
- we are only writing one byte in a `write` call
- the FD is not a regular file, socket or pipe
- the FD is marked as non-blocking

Please read the man page and correct me if I got it wrong.

Reply:
I think we're good with current code.

Reply:
I checked the `Ok(0)` behavior by adding a special metric which was incremented on the `Ok(0)` match arm. It was always 0 in all test runs.